cmake/all.cmake

```diff
@@ -128,6 +128,7 @@ foreach(filename
     list_append_deduplicate
     list_append_unique
     list_insert_in_workspace_order
+    list_insert_options
```

Comment:
I am not in favor of a particular name.
I have chosen "insert" because of the set logic, but if you like `list_append_unique_options` better that's fine for me as well.